web/modules/custom/dpl_favorites_list_material_component/src/Controller/DplFavoritesListMaterialComponentController.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Drupal\dpl_favorites_list_material_component\Controller;
+
+use Drupal\Core\Block\BlockManagerInterface;
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Render\RendererInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+
+/**
+ * Render favorites list material component react app.
+ */
+class DplFavoritesListMaterialComponentController extends ControllerBase {
+
+  /**
+   * DplFavoritesListMaterialComponentController constructor.
+   *
+   * @param \Drupal\Core\Block\BlockManagerInterface $blockManager
+   *   Drupal block manager.
+   * @param \Drupal\Core\Render\RendererInterface $renderer
+   *   Drupal renderer service.
+   */
+  public function __construct(
+    private BlockManagerInterface $blockManager,
+    private RendererInterface $renderer
+  ) {
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *   The Drupal service container.
+   *
+   * @return static
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('plugin.manager.block'),
+      $container->get('renderer'),
+    );
+  }
+
+  /**
+   * Display page with the favorites list material component.
+   *
+   * @return mixed[]
+   *   Render array.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   */
+  public function list(): array {
+    $config = [];
+
+    /** @var \Drupal\dpl_favorites_list_material_component\Plugin\Block\FavoritesListMaterialComponentBlock $plugin_block */
+    $plugin_block = $this->blockManager->createInstance('dpl_favorites_list_material_component_block', $config);
+
+    $access_result = $plugin_block->access($this->currentUser());
+    if (is_object($access_result) && $access_result->isForbidden() || is_bool($access_result) && !$access_result) {
+      throw new AccessDeniedHttpException();
+    }
+
+    // Add the cache tags/contexts.
+    $render = $plugin_block->build();
+    $this->renderer->addCacheableDependency($render, $plugin_block);
```

Comment:
How is this connected to the configuration form?
Will the cache be invalidated when settings are changing?

Reply:
If I understand it, it think it will be, but it's not 110% clear. It's an standard pattern for rending blocks that we found on drupal.org.

Reply:
It wont be invalidated when config is updated. It is easy to test though. Update config and see if varnish is being refreshed. You need to make the rendering aware of cache tags.

We have been handling settings and cache invalidation it in different ways through the modules. I have been trying to standardise it in the work with the instant loan settings.
It should be documented I think, but here it comes...

What you need to do is rather simple:
Create a [settings object ](https://github.com/danskernesdigitalebibliotek/dpl-cms/blob/c2af21a4982cdaa2e0e812975eedce9cd31d6b67/web/modules/custom/dpl_instant_loan/src/DplInstantLoanSettings.php#L10) that extends `DplReactConfigBase`.

Make sure to implement the `getConfigKey()` in the settings object which will be used in the settings form:
[Here](https://github.com/danskernesdigitalebibliotek/dpl-cms/blob/c2af21a4982cdaa2e0e812975eedce9cd31d6b67/web/modules/custom/dpl_instant_loan/src/Form/DplInstantLoanSettingsForm.php#L51), [here](https://github.com/danskernesdigitalebibliotek/dpl-cms/blob/c2af21a4982cdaa2e0e812975eedce9cd31d6b67/web/modules/custom/dpl_instant_loan/src/Form/DplInstantLoanSettingsForm.php#L66) and [here](https://github.com/danskernesdigitalebibliotek/dpl-cms/blob/c2af21a4982cdaa2e0e812975eedce9cd31d6b67/web/modules/custom/dpl_instant_loan/src/Form/DplInstantLoanSettingsForm.php#L118).

In your rendering object (block or controller) [inject the settings object](https://github.com/danskernesdigitalebibliotek/dpl-cms/blob/c2af21a4982cdaa2e0e812975eedce9cd31d6b67/web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php#L48).
And [this](https://github.com/danskernesdigitalebibliotek/dpl-cms/blob/c2af21a4982cdaa2e0e812975eedce9cd31d6b67/web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php#L346) makes the rendering aware of the settings in the rendering context.

Reply:
Yes, I will merge this PR then and make a new one with the pattern applied to the different modules from ITKDev